### ISSUE TITLE
Captcha and US Sizes Issues

### ISSUE THREAD

**Issue by xdmod2** [kdcz/3s_atc#5]
The captcha seems extremely slow and you need to solve it 3-4 times before it go through. I just notice the captchas are always disappearing image only. After successfully solving the captcha, I got an "Error! Check logs" status, and "no sizes available although there's a plenty of sizes are available at Adidas.com.

**Comment by jalex1** [comment]
^^^This is the exact same issues that I have (for co.uk)

**Comment by kdcz** [comment]
Do you use chrome to solve captcha ? Because thats why I implemented captcha solving in Chrome, the captcha browser in the "Tools" tab is an IE browser(and I can't change it atm) so for some reasons it is very slow and doesn't solve the first time.

**Comment by kdcz** [comment]
@jalex1 I solved the problem for UK but weird thing that it doesn't works on US, i'll have a look at it

**Comment by aagulto** [comment]
![capture](https://cloud.githubusercontent.com/assets/20647076/22937668/3553bbca-f2a7-11e6-8994-577865951018.JPG)

**Comment by aagulto** [comment]
I **always** use Chrome browser.

**Comment by kdcz** [comment]
Going to look at it

**Comment by kdcz** [comment]
Should be fixed can you try again please ? Thank you

**Comment by aagulto** [comment]
I can't resolved the captcha at all. It's extremely slow (disappearing images) and have tried to solved it 10 times, no luck.

**Comment by aagulto** [comment]
I'm not sure if my work computer firewall comes into play, I can't turn off my firewall at work due to GPO settings. I'll try it later when I get home.

**Comment by kdcz** [comment]
I was talking about the size checking...
Nevermind I pushed an update but not sure if it will fix something can you check ?

**Comment by aagulto** [comment]
But I can't add the sizes without solving the captcha first?

**Comment by kdcz** [comment]
You can solve captcha even if the program doesn't asks you to, just right click and 'Solve captcha'

**Comment by aagulto** [comment]
I did, captcha is extremely slow and cannot be solve the puzzle even I've tried to solved it so many times. I using the same key with SM script which working without any issue.

**Comment by aagulto** [comment]
For your reference:
![capture](https://cloud.githubusercontent.com/assets/20647076/22980984/1154d5e8-f369-11e6-9779-80d4c44f218a.JPG)

**Comment by kdcz** [comment]
Did you replace the old sitekey.php with the new I just uploaded?

**Comment by aagulto** [comment]
You know what... I forgot to replace that. Hold on, let me try it real quick!

**Comment by aagulto** [comment]
Nope, same thing! :-/

**Comment by aagulto** [comment]
https://www.dropbox.com/s/l7fjkojv50phyxv/IMG_2519.MOV?dl=0

**Comment by aagulto** [comment]
Same thing, see how slow the captcha?

**Comment by kdcz** [comment]
That's weird because I don't have this problem, can we setup a teamviewer so I can see what's wrong ?

**Comment by aagulto** [comment]
I'm a work right now and trying to share you what's the issue as much as possible. Probably later when I get home but time zone will be a big problem I guess?

**Comment by kdcz** [comment]
Do you have Skype? it would be faster to communicate

**Comment by aagulto** [comment]
I don't have an Skype nor Teamviewer but I'll be more than happy to set it up later when I get home. Kinda hectic right now at work, my apologies.

**Comment by kdcz** [comment]
No problems I'll wait, just send me your username on reddit so I can add you ;)

**Comment by kdcz** [comment]
Do you have the same problem with SM's script on the same device?

**Comment by aagulto** [comment]
Bro, I'm the same person who's offered you for help last time? Remember? Check your Reddit inbox, we have recent convo there. Yes, I'm using SM script with same parameters without any issue.

**Comment by kdcz** [comment]
Sorry I have a lot of people sending me PM's about the bot can't remember everyone :') 

Can you try this sitekey.php ? thank you 
http://pastebin.com/BMifDg5z

**Comment by kdcz** [comment]
Also you don't need to use the bot to try the sitekey.php file, just open you browser and type http://dev.adidas.com/sitekey.php?key=sitekeyhere

**Comment by aagulto** [comment]
Okay, so I have made couple of tests:

- First Test - Updated the sitekey from pastebin and solve the captcha without using the bot. Working like a champ! The captcha was resolved pretty quickly.

- Second Test - Solving the captcha through bot. Same problem, very slow and can't solve the puzzle.

I think the problem is from 3s_atc bot itself, and not from the sitekey.php. Hope this helps!

**Comment by kdcz** [comment]
Oh okay it makes sense, do you use Chrome aswell outside the bot ?

**Comment by aagulto** [comment]
Yes, Chrome is my main browser for everything.

**Comment by kdcz** [comment]
So I just figured out that it was not my fault but selenium's one(library I used to launch chromedriver and handle the captcha's response) : for some reasons it was slowing down the recaptcha api.So I had to completely change that chrome captcha system : now it uses normal chrome browser and checks in the cookies sql database of chrome for the response cookie.

Can you try now please? Thank you(don't forget to update also the sitekey.php)

**Comment by aagulto** [comment]
That's a good findings! So updated my version and everything seems to be working fine again. However, why do I need to close the Chrome entirely to trigger the ATC? "Solved captcha, please close your browser so the application can access to the recaptcha response." I don't think it's feasible...

**Comment by kdcz** [comment]
I'm sorry for this but yes you have to, the reasons are that I didn't find a better way of getting the captcha response and the actual method sends a sql command to the Chrome's cookie database to get the captcha cookie and I can't access the database if Chrome is opened because the database is locked. So I guess for now its the only way I can get the response or I can replace chrome by IE until I find a new solution.

**Comment by aagulto** [comment]
No worries, it is what it is. IMHO, closing the Chrome during the Zebra release next week is definitely not a good idea :-/. I'll wait for the future releases and hopefully you'll find a feasible solutions.

**Comment by jalex1** [comment]
I updated to the latest version of the script but now seems unable to log in or find inventory values? This is for adidas.co.uk

**Comment by kdcz** [comment]
Tested on adidas.co.uk logging in + inventory checking and works just fine

**Comment by jalex1** [comment]
Well i can log in on the browser, but no in the bot?  The phantonjs is in the same folder as the script and there is no firewall stopping any connections ?

**Comment by kdcz** [comment]
Did you wait at least 1 minute before notifying me that there is a problem ?

**Comment by jalex1** [comment]
Yes

**Comment by jalex1** [comment]
waited 30mins

**Comment by kdcz** [comment]
I can't really know what's wrong because everything works just fine for me, do you have Skype so I can figure out whats wrong?

**Comment by kdcz** [comment]
Nevermind, my bad I forgot to put some DLL's to the root folder, sorry again it should work now!

**Comment by jalex1** [comment]
Yeh, its now able to log on and the captcha is much quicker and easier. Good job! However, now the size issue is back, it is unable to find sizes, 'error check logs' 'no sizes available' when there are. The inventory checker also says there are. So an error in the main 3s_atc? (Again this is for adidas.co.uk)

**Comment by kdcz** [comment]
Thank you for reporting thing, I'll check that

**Comment by kdcz** [comment]
@aagulto Pushed a new update : you don't need to close your browser anymore you can even keep the tab open, don't forget to update your sitekey.php with the new one and update your settings in the application!

@jalex1 can you try again with the new update ? and make sure your locale is adidas.co.uk

**Comment by jalex1** [comment]
Thanks for the update, ran the new version and used the new sitekey. Seems to be unable to log in or connect now as it is unable to find any inventory values for any products 'getting inventory for product ...' and is unable to log in 'logging in'. I am able to log in to adidas.co.uk on my browser and there is no firewall.

**Comment by rizzryan** [comment]
Sorry I am new to this but how to set it up on Mac?

**Comment by aagulto** [comment]
@kdcz the new version works like a charm! Captcha is much quicker now. Thank you so much for fixing the issue :+1:

**Comment by jalex1** [comment]
Found the problem on mine, was my fault and sorted now. Works great!

**Comment by jalex1** [comment]
Do you think this will work for the yeezy release on the 25th? With hmac and if so, how should we set the profile up for it?

**Comment by jalex1** [comment]
Also just ran some more tests, seems to work for some products but no others and gets stuck on 'checking sizes' or 'no sizes found'.

**Comment by kdcz** [comment]
-Nice what was your problem ?
-For the yeezys just put the product ID,Email,Password and sizes by order of preference,the program will get the sitekey,client id,duplicate and hmac by himself 
-Can you give me an example of a product that doesnt works ?Thank you

**Comment by jalex1** [comment]
For example, by2747 gets stuck saying 'checking sizes' and others like C77124 and BB1100 say 'error check logs' 'no sizes available'. But the inventory checker works for them all and says there are. Only got it to work once with the by2747. So seems like there is a problem around finding sizes on the main bot? 

Also for the yeezy release, to get more than 1 will i need to have more than one profile, with a differnt account? 
Thanks again for all your work

**Comment by kdcz** [comment]
That is really weird I'll check that.
-For the yeezys you will need one more profile with different account indeed otherwise adidas will cancel your orders so I recommend you to use a different name aswell maybe one of your family?
But don't worry once a proxy reached the product page all profiles should cart

**Comment by jalex1** [comment]
Okay thanks and is there any way to test it on a splash page before the release or not?

**Comment by kdcz** [comment]
I'll post here a test php file that simulates a splash page later when Ill be home

**Comment by kdcz** [comment]
https://3satc.000webhostapp.com/
there you go

**Comment by jalex1** [comment]
Thanks dies that use hmac? And any word on the no sizes avaliable issue?

**Comment by kdcz** [comment]
I'll fix that when I have time

**Comment by kdcz** [comment]
Should be fixed now can you try again please ? Thanks

**Comment by jalex1** [comment]
Updated it, still getting the same 'no sizes available error'

**Comment by kdcz** [comment]
damn I was almost sure it would work this time, I tested the bot with different products and for each zone(US,UK, EU) and worked perfectly, can you give me the product ID ?

**Comment by jalex1** [comment]
It seems to work sometimes but mostly not. For example i tested it on s82237 and it did not work but then i did it again later and it did, then again and it got suck on 'checking sizes'? Also did not work on either 'M20324' or S75075. I dont think its related to the product itself.

**Comment by kdcz** [comment]
Okay thanks for your help Ill work on it soon

**Comment by CashHash123** [comment]
I am at this same stage, @jalex1.
 when it does say added to cart, have you checked cart to find no product.

i have got this issue also, have not been able to add even testing without product with captcha 

Also, is there a way to use splash without proxy?

Thanks .

**Comment by kdcz** [comment]
Make sure to clear you adidas cookies(or use a different browser) and login again then go to your cart 

I dont see the point of using splash without proxy just open your browser and wait on splash page

**Comment by CashHash123** [comment]
Yup, opening cart in incognito showed products in cart.

Only other issues I have you already addressed is,
1. The freeze on checking sizes after submitting captcha on US, 
2. some products giving the size not found error.

Thanks for all the hard work and quick responses!  Much appreciated

**Comment by kdcz** [comment]
Yes I already said that I'll work on a fix soon

**Comment by kdcz** [comment]
Pushed a new update for the sizes issues, can you try again ? thank you

**Comment by CashHash123** [comment]
I have tested this on a product with captcha on ADC US and have been able to successfully add to cart

**Comment by jalex1** [comment]
Hey updated the file, tested the first product and was successful. I then tried 2 different products and has the same error as before. I then tried the 1st product again, where it got stuck on 'checking sizes'. And whenever i try to cart the one that succeed the 1st time it gets stuck on that, which is strange? Was thinking that maybe there is some memory cache so after i cart one, its stopping me from carting again? Hope that helps haha

**Comment by jalex1** [comment]
Uninstalled it and re downloaded it, got the same error straight away aswell

**Comment by kdcz** [comment]
@jalex1 have you a skype account or reddit ? So i can send you tests version of the bot because I'm not getting these errors

**Comment by jalex1** [comment]
Yeh my username on reddit is Jalex11

**Comment by kdcz** [comment]
@jalex1 check your pms!

**Comment by jalex1** [comment]
Tried using the new hmac bypass with proxies. Getting 'error' after proxy 'setting up' when using the test splash page. Anyone else getting this?

**Comment by jalex1** [comment]
Trying the new bypass option, proxies seem to get to the 'on splash page' but never pass the hmac or get the sitekey? This is testing on your test page

**Comment by kdcz** [comment]
Yes this is normal because my website is not a replica of adidas's splash page, mine just lets you in after 15 seconds so if you refresh the page, you're back to 0 and have to wait 15 seconds again but I did some tsts on my side and it is working fine, please note also that this option DOESN'T ASSURE A SPLASH BYPASS but tries to.

PS: check your reddit PM's

**Comment by jalex1** [comment]
So using the bypass option gives a better chance of getting through?

**Comment by kdcz** [comment]
Yes past releases have revealed that somehow you have better chance to reach the product page or pass the splash page by clearing cookies and refreshing until you reach the product page(see AAR)

**Comment by jalex1** [comment]
One of the proxies managed to get the hmac in the end?

**Comment by kdcz** [comment]
I guess 15 seconds is a bit short

**Comment by jalex1** [comment]
@kdcz sent you a link to the video on reddit

**Comment by kdcz** [comment]
Pushed a new udapte, check it out please

**Comment by jalex1** [comment]
Cheers for the update man, just ran it a couple of times on different products and unfortunately getting the same issues

**Comment by kdcz** [comment]
The only way I can help you is on teamviewer, so if you're okay send me a PM on reddit

**Comment by kdcz** [comment]
Pushed an update, please update your WHOLE folder and give me some feedbacks please

**Comment by jalex1** [comment]
Same issue still occurs after updating folder

**Comment by jalex1** [comment]
Took the L on the yeezy drop, got an error on the proxies, so none made it to the splash page

**Comment by jalex1** [comment]
Also heard that adidas have patched the bypass method

**Comment by kdcz** [comment]
Adidas blocked proxies on EU and indeed patched refresh method

**Comment by jalex1** [comment]
So its looking like there not much hope in the way of bots atm against splash pages

**Comment by kdcz** [comment]
There will always be workarounds to increase chances bypassing the splash page.
Going to implement a multi session from a single IP feature

**Comment by jalex1** [comment]
Okay sounds good, great work!

**Comment by kdcz** [comment]
Check new update, size checking should be fixed for good now with new login feature added so you can login even if the stock is not live yet so you can cart faster then! First press login button then run.
+fixed bugs when having multiple profiles with same email address
+fixed bugs when multiple profiles need a captcha

**Comment by jalex1** [comment]
Just ran the update:
Great work seems like that size issue has been resolved at last haha. Does this update include the multisession single ip feature for splash? 
Also, if for a relese with no splash, like the triple black, can we use this bot if we dont have the sitekey or client id? Or do we have to find that first on the adidas site?
Thanks for the great work!

**Comment by Jeisonx3** [comment]
For non splash page product you only need site-key to cart you don't need clientdID, as far as i know clientID is needed for yeezys and splash page products. Non splash page product as of now all have the same site-key. which is 6Le4AQgUAAAAAABhHEq7RWQNJwGR_M-6Jni9tgtA.

**Comment by jalex1** [comment]
Thanks for that, however im using that sitekey but keep getting the {"result":"ERROR","basket":[]}
error

**Comment by kdcz** [comment]
@jalex1 for non splash products, the sitekey and clientid is the same as before usually but if I have time, I'll implement a system to reach the product page and read the sitekey, clientid...
@Jeisonx3 is right is right, ClientID and duplicate are not really needed for non splash products, only sitekey for the captcha.
Just pushed a new update for that multi session thing, check the settings page.
{"result":"ERROR","basket":[]} = wrong sitekey(since the last yeezy the sitekey has changed so check it) or wrong/missing captcha response

**Comment by jalex1** [comment]
Yeh I found a new sitekey that worked 6LeOnCkTAAAAAK72JqRneJQ2V7GvQvvgzsVr-6kR . And that would be very useful as you already have it for splash mode (retreiveing sitekey ect).
Ill have a look at the multisession feature

**Comment by Jeisonx3** [comment]
Oh yeah that site-key you found is for UK products.

**Comment by jalex1** [comment]
Just had a look at the multisession featute, could you explain what the 'number of sessions' and 'number of refresh sessions' are? Thanks

**Comment by jalex1** [comment]
Oh and whats the differences in splash modes exactly?

**Comment by kdcz** [comment]
Number of sessions : how many browser you wants to stay on splash page with a different session for each of them.
Number of refresh session : same but uses the 'old' refresh method to bypass splash(and so you have a new session each 30seconds for these browsers.

Does that method still works ? It wont bypass the splash instantly like it could before, but mrPorter on Reddit has figured out that on Alexander Wang's new collection help page,it says that staying on splash page longer doesnt increase your chances.Btw sessions can get you banned but I dont think so because else It would ban the public wifis,hotspots and large networks like universities,...

Difference between splash mode : -proxies = the previous method of using multi proxies, if you have proxies that didnt get banned from adidas website 
-multi-sessions : open multiple browser with different sessions with your ip

**Comment by jalex1** [comment]
So if i put number of sessions=5, would that be 5 sessions per ip? And would it then make sense to make the number of refresh sessions=5 also?

**Comment by kdcz** [comment]
Not per IP but with YOUR IP address 
Yes it will make sense,you'll have 10 sessions, 5 for each method

**Comment by kdcz** [comment]
Just pushed a new update, you can now transfer sessions : right click on session when it reached the product page and click "transfer session", it will open a new chrome browser with the product page.
Sessions worked fine on today's Alexander Wanger release that had splash page, past the splash with one session.

Also, here you got a nice splash page for testing purpose, it just doesn't have hmac but it doesn't matters you can put it to your profile : http://adidas.bot.nu/yeezy (thanks to https://www.reddit.com/user/M0rbz)

**Comment by jalex1** [comment]
Nice one, so did you cop? Also just to clarify at what point would we want to transfer the session?

**Comment by kdcz** [comment]
I didnt because it was on US website and Im in EU, and the shoe is so ugly to me lmao.

Well if for x reasons the program doesnt cop to your account, you can just transfer the session how many times you want and cop manually

**Comment by kdcz** [comment]
And it can be usefull aswell if the splash page is different from the yeezys one like today's splash : when you reached the product page you had to select 1 of the 4 products that redirected you to another page so the program couldnt get the sitekey,clientid,duplicate ,in this situation you can justen transfer session and cop.

**Comment by jalex1** [comment]
Ah great thanks

**Comment by Jeisonx3** [comment]
Hey i got a couple of suggestions, first can a delete button be added to delete profiles and added proxies, Also when i click run my accounts login in but i have to click run again for it to start the process of carting (if i just wanted to login i will click the login button) it think it will be better if the run button will login and starts the all task. Last can a harvest token be implemented or instead of going back to the bot and clicking each profile and opening a chrome tab to solve a captcha for each profile , can one chrome tab solve the captcha for all active profiles?

**Comment by jalex1** [comment]
@Jeisonx3 for the proxies just click on them then press delete on your keyboard?

@kdcz I have a quick suggestion, any chance you could make it so the size selections are saved. So dont have to put them all in again for each profile?

**Comment by kdcz** [comment]
@Jeisonx3 like @jalex1 said just select them and press delete key on you keyboard
Regarding the "run" button,it does login and run the taks but I implemented the "login" button so you can login before the product goes live and so cart faster.
For the captcha thing,I will correct this so you can keep 1 tab only without having to right click 

@jalex1 they aren't? They should be saved

**Comment by jalex1** [comment]
Yeh, i always have to put them in again for each time I make a new profile

**Comment by Jeisonx3** [comment]
@kdcz xD i tried that doesn't work i'm on a mac using windows via parallels doesn't seem to work ill figure it out it's probably in my keyboard setting.

**Comment by kdcz** [comment]
@jalex1 thats not normal I'll check that later 
@Jeisonx3 oh okay,will do then...

**Comment by jalex1** [comment]
How many sessions would you recommend to use? And also if one session gets past the splash page, does it cart for all profile?

**Comment by kdcz** [comment]
@jalex1 It all depends on your RAM, I would start with 10 then make some tests with that splash test I just posted to see how much sessions your computer can support. If one session gets past the splash page, it will cart for all profiles as always.

**Comment by magboo** [comment]
stocked on logged in..
![capture](https://cloud.githubusercontent.com/assets/25304953/23744608/2ccda644-04b5-11e7-8d96-3b78b44088c7.JPG)

**Comment by kdcz** [comment]
If you pressed "Login" button you have to press "Run" after

**Comment by magboo** [comment]
i see..got another issue im sotck on logging in on  adidas.it
![capture](https://cloud.githubusercontent.com/assets/25304953/23795615/cc89b416-0596-11e7-8622-e25abf80965a.JPG)

**Comment by kdcz** [comment]
Connect on adidas.it and check if you are not temporarily banned from adidas website(because of logging spam) else it should be a problem on your side, check your firewall and/or your anti virus

**Comment by Universalx182x** [comment]
What does www./htdcos folder not found, update your settings mean?

Which folder is this?
Is required in case of solving captachas.

Thank you!

**Comment by Universalx182x** [comment]
Think i got the answer already in the readme. 
Just ignore my previous post.

TY

**Comment by Universalx182x** [comment]
Hi,

I have tested on the US site but got following issue:
Submitting the captcha in chrome didn't work, I always had to resolve.

After debugging I saw that it is hold in following line:

 **_while (!ReadCookie("dev.adidas.com", "g-recaptcha-response", ref captcha_response))
                System.Threading.Thread.Sleep(1000);**

Following line will not be reached, because it stays in the while loop.
            **captchas.Add(new C_Captcha { sitekey = profile.Sitekey, response = captcha_response, profileID = profile.index });_**

Could you please check?

Thank you!
